errors.js

```diff
@@ -1,18 +1,52 @@
 'use strict'
 
-const createError = require('errno').create
-const LevelUPError = createError('LevelUPError')
-const NotFoundError = createError('NotFoundError', LevelUPError)
+function createError (type, Proto) {
+  const Err = function (message, cause) {
+    if (typeof message === 'object' && message !== null) {
+      // Can be passed just a cause
+      cause = cause || message
+      message = message.message || message.name
+    }
+
+    message = message || ''
+    cause = cause || undefined
```

Comment:
This check would be a noop except if cause is something falsy and not `undefined`. Since we control error creation, what about making this code stricter too?

Reply:
https://github.com/Level/errors/issues/38